bon-macros/src/builder/builder_gen/build_from.rs

```diff
@@ -0,0 +1,133 @@
+use crate::builder::builder_gen::{BuilderGenCtx, member::Member};
+use crate::util::prelude::*;
+use proc_macro2::Span;
+use quote::quote;
+use syn::{Type, spanned::Spanned};
+
+pub(super) fn emit(ctx: &BuilderGenCtx, target_ty: &Type) -> Result<TokenStream> {
+    let mut tokens = TokenStream::new();
+
+    let field_vars: Vec<_> = ctx
+        .members
+        .iter()
+        .map(|member| {
+            let ident = member.orig_ident();
+            let ty = member.norm_ty();
+            let default_expr = quote! { ::core::default::Default::default() };
+
+            match member {
+                Member::Field(_) | Member::StartFn(_) => quote! {
+                    let #ident: #ty = self.#ident;
+                },
+                Member::Named(member) => {
+                    let index = &member.index;
+                    quote! {
+                        let #ident: #ty = match self.__unsafe_private_named.#index {
+                            Some(value) => value,
+                            None => from.#ident.clone(),
```

Comment:
The idea is that `build_from` accepts `T` by value and does not call `clone()` at all (that's why this method's name is a bit shorter since it is less expensive and should be the preferred thing), but `build_from_clone` accepts `&T` and calls `.clone()` if needed (that's why it has "clone" in its name).